frontend/src/components/VariantListPage/VariantListVariants.tsx

```diff
@@ -100,6 +119,46 @@ const VariantListVariants = (props: VariantListVariantsProps) => {
     ? variants
     : combineVariants(variants, structural_variants);
 
+  type DisplayNames = {
+    A: string;
+    B: string;
+    C: string;
+    D: string;
+  };
+
+  const handleEditTags = () => {
+    onOpen();
+  };
+
+  const handleSaveDisplayNames = () => {
+    const updatedGroups = { ...taggedGroups };
+
+    (Object.keys(editDisplayNames) as Array<keyof DisplayNames>).forEach(
```

Comment:
Nice, this is a very valid use of `as`, this inability of TS to get this type is a known quirk of TS having to deal with JS, so using `as` here makes a lot of sense.